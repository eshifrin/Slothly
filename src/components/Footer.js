import React, { Component } from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { colors } from '../constants';

export default ({ stats }) => (
  <View style={styles.container}>
    <Text style={styles.text}>{`TOTAL: ${stats.minutesSoFar || '...'}`}</Text>
    <Text style={styles.text}>{`YEARLY PACE: ${stats.yearlyPace ||
      '...'}`}</Text>
  </View>
);

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    height: 30,
    justifyContent: 'space-around'
  },
  text: {
    color: colors.deepBlue,
    fontSize: 20,
    fontFamily: 'futura',
    textAlign: 'center'
  }
});
