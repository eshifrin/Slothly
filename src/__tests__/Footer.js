import 'react-native';
import React from 'react';
import Footer from '../components/Footer';
import { shallow } from 'enzyme';

describe('Footer', () => {
  let wrapper;
  const mockStats = {
    minutesSoFar: '65',
    yearlyPace: '100'
  };

  beforeEach(done => {
    wrapper = shallow(<Footer stats={mockStats} />);
    setTimeout(() => done(), 0);
  });

  it('renders', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
