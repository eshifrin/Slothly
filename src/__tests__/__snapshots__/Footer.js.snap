// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "height": 30,
      "justifyContent": "space-around",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#032b82",
        "fontFamily": "futura",
        "fontSize": 20,
        "textAlign": "center",
      }
    }
  >
    TOTAL: 65
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#032b82",
        "fontFamily": "futura",
        "fontSize": 20,
        "textAlign": "center",
      }
    }
  >
    YEARLY PACE: 100
  </Text>
</View>
`;
